crates/aranya-client/src/aqc/net.rs

```diff
@@ -0,0 +1,824 @@
+//                             receive_ctrl_message(&daemon, &mut channel_map, &mut conn).await
+//                         {
+//                             continue;
+//                         }
+//                     // If the PSK identity hint is not the control PSK, check if it's in the channel map.
+//                     // If it is, create a channel of the appropriate type. We should have already received
+//                     // the control message for this PSK, if we don't we can't create a channel.
+//                     } else if let Some(channel_info) = channel_map.get(identity) {
+//                         tracing::debug!(
+//                             "Found channel info in map for identity hint {:02x?}: {:?}",
+//                             identity,
+//                             channel_info
+//                         );
+//                         if let ControlFlow::Break(_) =
+//                             create_channel_type(&sender, conn, channel_info).await
+//                         {
+//                             return;
+//                         }
+//                     } else {
+//                         tracing::debug!(
+//                             "No channel info found in map for identity hint {:02x?}",
+//                             identity
+//                         );
+//                         continue;
+//                     }
+//                 } else {
+//                     tracing::warn!("No identity hint received. Unable to create channel.");
+//                 }
+//             }
+//             None => {
+//                 debug!("Server connection terminated");
+//                 break;
+//             }
+//         }
+//     }
+// }
+
+fn create_channel_type(conn: Connection, channel_info: &AqcChannel) -> AqcChannelType {
+    match channel_info {
+        AqcChannel::Bidirectional { id } => {
+            // Once we accept a valid connection, let's turn it into an AQC Channel that we can
+            // then open an arbitrary number of streams on.
+            let channel = AqcBidirectionalChannel::new(LabelId::default(), *id, conn);
+            AqcChannelType::Bidirectional { channel }
+        }
+        AqcChannel::Unidirectional { id } => {
+            // Once we accept a valid connection, let's turn it into an AQC Channel that we can
+            // then open an arbitrary number of streams on.
+            let receiver = AqcReceiverChannel::new(LabelId::default(), *id, conn);
+            AqcChannelType::Receiver { receiver }
+        }
+    }
+}
+
+async fn receive_ctrl_message(
+    daemon: &Arc<DaemonApiClient>,
+    channel_map: &mut HashMap<Vec<u8>, AqcChannel>,
+    conn: &mut Connection,
+) -> Result<(), AqcError> {
+    match conn.accept_bidirectional_stream().await {
+        Ok(Some(stream)) => {
+            let (mut recv, mut send) = stream.split();
+            if let Ok(Some(ctrl_bytes)) = recv.receive().await {
```

Comment:
We can reduce nesting levels with this alternate let pattern:
```suggestion
            let Ok(Some(ctrl_bytes)) = recv.receive().await else {
                error!("Failed to receive control message or stream closed");
                return Err(AqcError::ConnectionClosed); // Or a more specific error
            };
```
Don't forget to remove the trailing else {} section and un-indent the match locally. You should able to use this in a handful of other places here, anywhere you simply return an Error (it has to alter codeflow to compile)